spring-cloud-sleuth-core/src/main/java/org/springframework/cloud/sleuth/instrument/quartz/TraceQuartzAutoConfiguration.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright 2013-2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.sleuth.instrument.quartz;
+
+import brave.Tracing;
+import org.quartz.Scheduler;
+
+import org.springframework.beans.factory.InitializingBean;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.autoconfigure.AutoConfigureAfter;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnBean;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.boot.autoconfigure.quartz.QuartzAutoConfiguration;
+import org.springframework.cloud.sleuth.autoconfig.TraceAutoConfiguration;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+
+/**
+ * {@link org.springframework.boot.autoconfigure.EnableAutoConfiguration
+ * Auto-configuration} enables Quartz span information propagation.
+ *
+ * @author Branden Cash
+ */
+@Configuration
+@ConditionalOnBean({ Tracing.class, Scheduler.class })
+@AutoConfigureAfter({ TraceAutoConfiguration.class, QuartzAutoConfiguration.class })
+@ConditionalOnProperty(value = "spring.sleuth.quartz.enabled", matchIfMissing = true)
+public class TraceQuartzAutoConfiguration implements InitializingBean {
+
+	private Scheduler scheduler;
+
+	private Tracing tracing;
+
+	@Autowired
+	public TraceQuartzAutoConfiguration(Scheduler scheduler, Tracing tracing) {
```

Comment:
I don't think we actually need the `@Autowired`. Also, throught the project AFAIR we're setting the beans via fields in the auto configurations.